### ISSUE TITLE
TransformerArchitecture layers= override yields 0 trainable parameters → ppl >>uniform

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1382]
## Summary

Passing a custom `layers:` list to `TransformerArchitecture<T>` via `AiModelBuilder.ConfigureModel(...).BuildAsync()` results in a trained model with **0 trainable parameters**, even when `numEncoderLayers >= 1` and `numHeads >= 1` are specified in the same architecture. Inference produces strongly mis-calibrated logits (ppl orders of magnitude worse than uniform softmax), not the silent #1380 mode-collapse-to-uniform signature.

This is **distinct from #1380**:
- #1380 (classical Transformer): top1 ≈ 1/V (~0.39% at V=256), ppl ≈ V (~256). Silent collapse to uniform.
- This issue: top1 = 0%, ppl ≫ V (e.g., **ppl = 83,589** at V=256). Confidently-wrong distribution.

## Empirical reproducer

HE side: `tests/Learning/Phase_PA_FacadeHRE_SmokeTest.cs` → `HRETransformer_Facade_BuildAsync_ModeCollapseCheck`.

Config:
- vocab=256, dModel=64, dFf=128, ctxLen=32, heads=2, layers=1, epochs=2
- 10KB WikiText-2 slice (real text, not random bytes)
- `TransformerArchitecture<float>(... numEncoderLayers: 1, numHeads: 2, modelDimension: 64, layers: [HreEmbed, HrrPE, HreReadout])`
- All three custom layers are zero-trainable-parameter (substrate-correct fixed phase codebook + Born readout)

Result (verified on AiDotNet 0.204.0 / Tensors 0.81.3):
```
Build OK in 3.52s. trainable_params = 0
   top1 = 0.00%  (0/200)
   ppl  = 83589.08  (uniform-softmax baseline = 256)
```

`_trained.TotalTrainableParameters` reports 0 even though `numEncoderLayers: 1, numHeads: 2` should have instantiated a `MultiHeadAttentionLayer<float>` with trainable Q/K/V/O projections.

## Hypotheses

1. **`layers:` REPLACES the entire auto-built stack** (not "insert as bridge layers"). When user provides a custom layer list, the Transformer constructor uses ONLY those layers and skips the auto-built encoder block. Since the HRE chain is zero-param, the resulting model has nothing to train.
2. **`TotalTrainableParameters` undercounts** in this composition: actual MHA exists but isn't counted, and Adam silently fails to update it. (Unlikely — ppl=83K suggests the MHA isn't running at all.)
3. **`MultiHeadAttentionLayer<float>` is constructed but not wired into the forward graph** when `layers:` is non-null.

Hypothesis 1 is the most consistent with the empirical signature.

## Why this matters

The HRE-substituted facade predictor (`AiDotNetFacadeHRETransformerLMPredictor` in HE) is the load-bearing apples-to-apples chain for PAPER-A's main claim — that HRE structural substitutions (L1 token embedding, L2 PE, L9 readout) strict-WIN against the classical Transformer on byte-LM perplexity. With 0 trainable parameters the chain literally can't learn anything; any "HRE wins" claim from this predictor is currently uninterpretable because both arms (HRE-facade and classical-facade via BuildAsync) are broken in different ways.

## Expected behavior

Either:
- (a) `layers:` should INSERT the custom layers as bridges around the auto-built attention stack (preserving the trainable MHA), with shape-validation at the seams; OR
- (b) `layers:` should REPLACE the stack but doc that explicitly + reject the call at build-time if `numEncoderLayers >= 1` was also specified (since the two arguments are mutually exclusive in that semantics); OR
- (c) `layers:` should REPLACE the stack but the user is responsible for providing their own trainable layers — in which case `TotalTrainableParameters = 0` should fail-fast with a clear error rather than silently building an untrainable model.

## Related

- #1380 — silent mode-collapse on classical Transformer via the same BuildAsync path. Per HE memory `reference_aidotnet_1380_facade_blocked.md`: classical-facade gives top1=0/ppl=256 vs per-sample top1=56/ppl=6.77 on the same config. Tracked via HE-side per-sample workaround.
- #1317 — closed AiDotNet 0.193: `layers:` constructor parameter now accepts substrate-correct layer lists with shape-based validation. The acceptance ≠ correct semantics if those layers replace the auto-built trainable head.
